backend/tests/integration-async/records/indexer/test_literature.py

```diff
@@ -383,3 +383,87 @@ def test_literature_regression_changing_bai_in_record_reindex_records_which_are_
         },
     ]
     retry_until_matched(steps)
+
+
+def test_versions_on_create_model(
+    inspire_app, celery_app_with_context, celery_session_worker
+):
+    import uuid
+
+    from invenio_records.api import Record
+    from invenio_records.models import RecordMetadata
+    from sqlalchemy.orm.attributes import flag_dirty, flag_modified
+
+    data = faker.record("lit", with_control_number=True)
+
+    rec = RecordMetadata(id=uuid.uuid4(), json=data)
+    data["titles"][0]["title"] = "UPDATED"
+    rec.json = data
+    db.session.add(rec)
+
+    data["titles"][0]["title"] = "UPDATED 2"
+    rec.json = data
```

Comment:
what if you copy `data`? I don't see how this could fail, given you're mutating the `dict` you've assigned previously.

Reply:
I'm not sure I understand, this test is passing as expected